### ISSUE TITLE
Circularity with multi-fasta input

### ISSUE THREAD

**Issue by asetGem** [gem-pasteur/Integron_Finder#43]
# Behaviour expected:

By default:
- if input is a fasta file, the single replicon is considered as circular
- if input is multi-fasta, all replicons are considered as linear

Available options:
- `--circ` used to set all replicons to circular (useful when multi-fasta provided. When fasta provided, it does not change anything from default...)
- `--linear` used to set all replicons to linear (useful when fasta input. When multi-fasta, it is already the default behaviour)
- `--topo-file <filename>`: the user can provide a file specifying the topology of the replicons not following the default or `--circ`/`--linear` option. 2 columns: replicon ID and `circular` or `linear`

To sum-up. For each replicon, its topology is:
- the one in topology file if given
- if not given in topology file, but  `--circ`/`--linear` option given, follow this option
- if not in topology file and no option given, use default (fasta->circular ; multi-fasta->linear)

Add replicon topology information in the final tab file.

# Steps

- [x] topology file parser to get topologies of given replicons if `--topo-file` given
- [x] add `--circ`, `--linear` and `--topo-file <filename>` options
- [x] error message from parser if `--circ` and `--linear` are given together
- [x] add column to final tab with replicon topology
- [x] Assign topology to all replicons according to the given options, and launch integron_finder with the appropriate topology

**Comment by bneron** [comment]
commit  ce40896 implements

- [x] topology file parser to get topologies of given replicons if `--topo-file` given
- [x] add `--circ`, `--linear` and `--topo-file <filename>` options
- [x] error message from parser if `--circ` and `--linear` are given together

but 8b38da1 install integron package needed for this feature

**Comment by bneron** [comment]
I have aquestion for the following feature

- [ ] add column to final tab with replicon topology

the topology will be set by --circ or --linear option or --topology-file
but in code I read
```
# If sequence is too small, it can be problematic when using circularity
    if len(SEQUENCE) > 4 * DISTANCE_THRESHOLD:
        circular = not args.linear
    else:
circular = False
```
so the topology set by the user can be override.
in the results what value must appear the topology set by the user or the topology effectively used?

**Comment by jeanrjc** [comment]
Well, I'm not sure. Ideally we shouldn't have such condition and let the algo behave normally. We should have a condition to stop in case the entire sequence is parsed or once an attC site is found for the second times, otherwise the expansion will never stops. 

I think we can keep it like this for now, and in the column use the value used by the algo. The true topology is in the topology file. We should name this column "Topology_considered" or something similar to stress the fact that it's not the topology of the sequence but the topology used by IF.

I think it's really an edge case, but we should mention it on the doc if it's not already the case. If it turns out that many people have very small plasmids with integron over the edge, we might implement a proper solution later, or maybe use a lower value than ` 4 * DISTANCE_THRESHOLD`.

TL;DR: Use the value of the parameter in the output file and not the value of the topology file, and rename the column to stress the fact that it's the value of the parameter, not the actual state of the sequence.

Thanks!
